pypeit/calibrations.py

```diff
@@ -964,6 +964,32 @@ def get_wv_calib(self):
             self.wv_calib = wavecalib.WaveCalib.from_file(cal_file)
             self.wv_calib.chk_synced(self.slits)
             self.slits.mask_wvcalib(self.wv_calib)
+            # TODO - should all of this new code be removed??
+            if self.par['wavelengths']['method'] == 'echelle':
+                msgs.info('Method set to Echelle -- checking wv_calib for 2dfits')
+                if not hasattr(self.wv_calib, 'wv_fit2d'):
+                    msgs.info('2d fit does not exist! Doing now: ')
+                    is_arc = self.fitstbl.find_frames('arc', calib_ID=self.calib_ID)
+                    lamps = self.spectrograph.get_lamps(self.fitstbl[is_arc]) \
+                                if self.par['wavelengths']['lamps'] == ['use_header'] \
+                                else self.par['wavelengths']['lamps']
+                    meta_dict = dict(self.fitstbl[is_arc][0]) \
+                                if self.spectrograph.pypeline == 'Echelle' \
+                                    and not self.spectrograph.ech_fixed_format else None
+                    waveCalib = wavecalib.BuildWaveCalib(self.msarc, self.slits, self.spectrograph,
+                                        self.par['wavelengths'], lamps, meta_dict=meta_dict,
+                                        det=self.det, qa_path=self.qa_path)
+
+                    fit2ds, dets, order_dets = waveCalib.echelle_2dfit(
+                        self.wv_calib, skip_QA = False, debug=False)
+                    # Save
+                    self.wv_calib.wv_fit2d = np.array(fit2ds)
+                    # Save det_img?
+                    if self.par['wavelengths']['ech_separate_2d']:
+                        self.wv_calib.det_img = self.msarc.det_img.copy()
+                    #update file with the new 2d fits
+                    self.wv_calib.to_file()
```

Comment:
There may be a better way to do this, but this was to patch an issue where the Echelle method assumes the WaveCalib file will have 2dfits in it without checking. So if for some reason the WaveCalib file does not have the 2dfit, now PypeIt will compute one, rather than crashing further along. The code could potentially be replaced by an error message just explaining that a valid 2d fit is necessary for the Echelle method.

Reply:
Yes, this should not be here.

Reply:
I've added an error message to warn the user and removed the code.